packages/vite/src/node/server/moduleGraph.ts

```diff
@@ -70,6 +70,19 @@ export class ModuleGraph {
     return this.fileToModulesMap.get(file)
   }
 
+  addWatchModuleToFile(file: string, id: string): void {
+    let fileMappedModules = this.fileToModulesMap.get(file)
```

Comment:
Sorry, this is also missing for Windows compat
```suggestion
  addWatchModuleToFile(file: string, id: string): void {
    file = normalizePath(file)
    let fileMappedModules = this.fileToModulesMap.get(file)
```

I'm checking if we should use `createFileOnlyEntry` and add the module directly in `transformRequest` so we don't need to go from module to id to module here and how this new file module interacts with other modules. There could be duplicated entries for this file if later `ensureEntryFromUrl` is called because the file is imported.